springfox-swagger2/src/main/java/springfox/documentation/swagger2/configuration/Swagger2JacksonModule.java

```diff
@@ -79,4 +86,64 @@ public void setupModule(SetupContext context) {
   private class CustomizedSwaggerSerializer {
   }
 
+  @JsonAutoDetect
+  @JsonInclude(value = Include.NON_EMPTY)
+  private interface PropertyExampleSerializerMixin {
+
+    @JsonSerialize(using = PropertyExampleSerializer.class)
+    Object getExample();
+
+    class PropertyExampleSerializer extends StdSerializer<Object> {
+
+      private final static Pattern JSON_NUMBER_PATTERN = Pattern.compile("-?(?:0|[1-9]\\d*)(?:\\.\\d+)?(?:[eE][+-]?\\d+)?");
+
+      @SuppressWarnings("unused")
+      public PropertyExampleSerializer() {
+        this(null);
+      }
+
+      PropertyExampleSerializer(Class<Object> t) {
+        super(t);
+      }
+
+      @Override
+      public void serialize(Object value, JsonGenerator gen, SerializerProvider provider)
+              throws IOException {
+        String stringValue = (value instanceof String) ? ((String) value).trim() : value.toString().trim();
+        if (isNotJsonString(stringValue)) {
+          gen.writeRawValue(stringValue);
+        } else {
+          gen.writeString(stringValue);
+        }
+      }
+
+      private boolean isNotJsonString(final String value) throws IOException {
+        // strictly speaking, should also test for equals("null") since {"example": null} would be valid JSON
+        // but swagger2 does not support null values
+        // and an example value of "null" probably does not make much sense anyway
+        return value.startsWith("{")                              // object
+                || value.startsWith("[")                          // array
+                || "true".equals(value)                           // true
+                || "false".equals(value)                          // false
+                || JSON_NUMBER_PATTERN.matcher(value).matches();  // number
```

Comment:
Yes, sure. But I'm going to need a little help with that.
I am not that familiar with groovy and the spock framework.
Also, where to put the tests? Can I extend some existing test class or create a new one?
Would be great if you could give me some starting point.